registration-processor/pre-processor/registration-processor-cmd-validator-stage/src/main/java/io/mosip/registration/processor/stages/cmdvalidator/DeviceValidator.java

```diff
@@ -88,42 +92,73 @@
 	 */
 
 	public void validate(RegOsiDto regOsi,String process, String registrationId)
-			throws JsonProcessingException, IOException, BaseCheckedException, ApisResourceAccessException, JSONException {
-
-		BiometricRecord biometricRecord = packetManagerService.getBiometricsByMappingJsonKey(registrationId,
-				MappingJsonConstants.INDIVIDUAL_BIOMETRICS, process,
-				ProviderStageName.CMD_VALIDATOR);
-		List<BIR> birs=biometricRecord.getSegments();
-		List<JSONObject> payloads=new ArrayList<>();
-		for(BIR bir: birs) {
-			if(bir.getOthers()!=null) {
-			for(Entry entry: bir.getOthers()) {
-				if(entry.getKey().equals("PAYLOAD")) {
-					payloads.add(new JSONObject(entry.getValue()));				
-				}
-			}
+			throws JsonProcessingException, IOException, BaseCheckedException, 
+				ApisResourceAccessException, JSONException {
+		List<String> fields = Arrays.asList(MappingJsonConstants.INDIVIDUAL_BIOMETRICS,
+				MappingJsonConstants.AUTHENTICATION_BIOMETRICS, 
+				MappingJsonConstants.INTRODUCER_BIO,
+				MappingJsonConstants.OFFICERBIOMETRICFILENAME, 
+				MappingJsonConstants.SUPERVISORBIOMETRICFILENAME);
+		for (String field : fields) {
+			String value = packetManagerService.getField(registrationId, field, process, 
```

Comment:
i can see u r adding officer and supervisor bio file also in fields. But officer and supervisor bio will be present in metainfo json not idjson. Please check file name in metainfo